dev/release/verify-release-candidate.sh

```diff
@@ -756,7 +756,7 @@ test_python() {
   show_header "Build and test Python libraries"
 
   # Build and test Python
-  maybe_setup_virtualenv "cython>=0.29.31" numpy "setuptools_scm<8.0.0" setuptools
+  maybe_setup_virtualenv "cython>=0.29.31" numpy "setuptools_scm>=8.0.0" "setuptools>=64"
```

Comment:
FWIW, for _release_ verification I think we should actually remove this line entirely, and rely on the fact that we specify those dependencies in pyproject.toml (i.e. replacing the `python setup.py build_ext --inplace` below with `python -m pip install (-e) .`